tests/integration/helpers.py

```diff
@@ -1,11 +1,53 @@
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+import json
 import logging
+import urllib.request
 
-log = logging.getLogger(__name__)
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
 
 
 async def get_unit_address(ops_test, app_name: str, unit_num: int) -> str:
     status = await ops_test.model.get_status()  # noqa: F821
     return status["applications"][app_name]["units"][f"{app_name}/{unit_num}"]["address"]
+
+
+async def is_loki_up(ops_test, app_name) -> bool:
+    address = await get_unit_address(ops_test, app_name, 0)
+    url = f"http://{address}:3100"
+    logger.info("Loki public address: %s", url)
+
+    response = urllib.request.urlopen(
+        f"{url}/loki/api/v1/status/buildinfo", data=None, timeout=2.0
+    )
+    return response.code == 200 and "version" in json.loads(response.read())
+
+
+class IPAddressWorkaround:
```

Comment:
Once again, I really don't like calling this `IPAddressWorkaround`. This is not at all related to ip addresses. It just shortens the update status interval.

Reply:
Depends how you look at it:
- for the purpose of this test I do not need to change the update status per se, what I need is an IP address
- whatever happens inside is an implementation detail; currently update status does the trick, tomorrow maybe something else. But ideally this bug would be fixed and the whole thing deleted.
In other words I prefer a semantic name over a technical one.